### PR TITLE
[firebase_auth_web] Add noop Android implementation.

### PR DESCRIPTION
**Pull request by ditman**
## Description

This is a mitigation for certain build paths that expect an android/ directory on every plugin (regardless of the platform)

## Related Issues

flutter/flutter#46898

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] If the pull request affects only one plugin, the PR title starts with the name of the plugin in brackets (e.g. [cloud_firestore])
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/FirebaseExtended/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/firebase_auth/firebase_auth_web/android/build.gradle [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+}
+
+apply plugin: 'com.android.library'
+
+android {
+    compileSdkVersion 28
+
+    defaultConfig {
+        minSdkVersion 16
+        testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
```
Comment (amirh) [inline packages/firebase_auth/firebase_auth_web/android/build.gradle]: delete this line

File: packages/firebase_auth/firebase_auth_web/android/gradle.properties [hunk 2]
```diff
@@ -0,0 +1,4 @@
+org.gradle.jvmargs=-Xmx1536M
+android.enableR8=true
+android.useAndroidX=true
```
Comment (amirh) [inline packages/firebase_auth/firebase_auth_web/android/gradle.properties]: delete these 2 lines

File: packages/firebase_auth/firebase_auth_web/android/src/main/java/io/flutter/plugins/firebaseauth_web/FirebaseAuthWebPlugin.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
+package io.flutter.plugins.firebaseauth_web;
+
+import androidx.annotation.NonNull;
```
Comment (amirh) [inline packages/firebase_auth/firebase_auth_web/android/src/main/java/io/flutter/plugins/firebaseauth_web/FirebaseAuthWebPlugin.java]: remove (and remove all @NonNull)

File: packages/firebase_auth/firebase_auth_web/CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,7 @@
+## 0.1.2
```
Comment (amirh) [inline packages/firebase_auth/firebase_auth_web/CHANGELOG.md]: Make this a patch version increase (0.1.1+1)

Comment (ditman) [reply packages/firebase_auth/firebase_auth_web/CHANGELOG.md]: I disagree, but we can discuss this in the post-mortem.

**Comment by ditman** [comment]
> Thanks, we've figured we need to make sure the no-op implementation does not reference AndroidX, left a few comments about that.

Done!

**Comment by ditman** [comment]
Waiting for build-ipas checks to pass.

**Comment by mklim** [comment]
For what it's worth I think in this case it would be okay to merge before the `build-ipas` tasks specifically finish. They're time consuming, this is a high priority issue, and nothing in this PR should theoretically affect iOS builds in any way.

**Comment by ditman** [comment]
> For what it's worth I think in this case it would be okay to merge before the `build-ipas` tasks specifically finish. They're time consuming, this is a high priority issue, and nothing in this PR should theoretically affect iOS builds in any way.

@mklim do hit "Squash and merge" if you feel it's OK to go!

**Comment by ditman** [comment]
Thanks @mklim!

File: packages/firebase_auth/firebase_auth_web/android/src/main/java/io/flutter/plugins/firebaseauth_web/FirebaseAuthWebPlugin.java [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+import io.flutter.plugin.common.PluginRegistry.Registrar;
+
+/** FirebaseAuthWebPlugin */
+public class FirebaseAuthWebPlugin implements FlutterPlugin {
+  @Override
+  public void onAttachedToEngine(FlutterPluginBinding flutterPluginBinding) {}
+
+  public static void registerWith(Registrar registrar) {}
+
+  @Override
```
Comment (amirh) [inline packages/firebase_auth/firebase_auth_web/android/src/main/java/io/flutter/plugins/firebaseauth_web/FirebaseAuthWebPlugin.java]: delete

File: packages/firebase_auth/firebase_auth_web/android/src/main/java/io/flutter/plugins/firebaseauth_web/FirebaseAuthWebPlugin.java [hunk 6]
```diff
@@ -0,0 +1,20 @@
+package io.flutter.plugins.firebaseauth_web;
+
+import io.flutter.embedding.engine.plugins.FlutterPlugin;
+import io.flutter.plugin.common.MethodCall;
```
Comment (amirh) [inline packages/firebase_auth/firebase_auth_web/android/src/main/java/io/flutter/plugins/firebaseauth_web/FirebaseAuthWebPlugin.java]: delete

**Pull request merged**